src/types.jl

```diff
@@ -59,26 +50,22 @@ function gen_cache(key::AbstractAESKey, mode=CBC)
 end
 
 # Ciphertext type
-"""
-"""
-mutable struct AESCipherText{ivType,modeType}
+mutable struct CipherText{ivType,mode}
 	data::Array{UInt8}
 	iv::ivType
 	keylength::Int
-	mode::modeType
-	original_type::Type
 end
+CipherText(data, iv::IV, keylength, mode::MODE) where {IV} = CipherText{IV,mode}(data, iv, keylength)
+const CT = CipherText
 
 # Cipher Type
-
-"""
-"""
-mutable struct AESCipher{mode,cacheType,keyType} <: AbstractCipher
+mutable struct Cipher{mode,cacheType,keyType} <: AbstractCipher
 	cache::cacheType
 	key::keyType
 end
+const AESCipher = Cipher
 
-function AESCipher(;key_length=128,mode=CBC,key=keygen(key_length))
+function Cipher(;key_length=128,mode=CBC,key=keygen(key_length))
```

Comment:
Don't change the type names, Cipher is too generic

Reply:
I've changed the type name internally only. Below the definition is `const AESCipher = Cipher` which is exported, so the API doesn't change. Internally, it's clear that everything is an AESCipher since we're in the AES package.

Reply:
Ohh makes sense